core/domain/user_domain.py

```diff
@@ -483,6 +493,10 @@ def unmark_banned(self, default_role: str) -> None:
         self.banned = False
         self.roles = [default_role]
 
+    def mark_lesson_info_modal_viewed(self) -> None:
+        """Sets user_has_viewed_lesson_info_modal_once to true."""
```

Comment:
"Sets user_has_viewed_lesson_info_modal_once to true which shows the user has viewed their progress through exploration in the lesson info modal at least once in their lifetime journey."

Reply:
Done!